sdk/servicebus/Azure.Messaging.ServiceBus/src/Administration/QueuePropertiesExtensions.cs

```diff
@@ -51,28 +53,32 @@ public static XDocument Serialize(this QueueProperties description)
         /// <summary>
         ///
         /// </summary>
-        public static QueueProperties ParseFromContent(string xml)
+        public static async Task<QueueProperties> ParseResponseAsync(Response response, ClientDiagnostics diagnostics)
         {
             try
             {
+                string xml = await response.ReadAsStringAsync().ConfigureAwait(false);
                 var xDoc = XElement.Parse(xml);
                 if (!xDoc.IsEmpty)
                 {
                     if (xDoc.Name.LocalName == "entry")
                     {
-                        return ParseFromEntryElement(xDoc);
+                        return await ParseFromEntryElementAsync(xDoc, response, diagnostics).ConfigureAwait(false);
                     }
                 }
             }
             catch (Exception ex) when (!(ex is ServiceBusException))
             {
-                throw new ServiceBusException(false, ex.Message);
+                throw new ServiceBusException(false, ex.Message, innerException: ex);
             }
-
-            throw new ServiceBusException("Queue was not found", ServiceBusFailureReason.MessagingEntityNotFound);
+            response.ContentStream.Position = 0;
+            throw new ServiceBusException(
+                "Queue was not found",
```

Comment:
Seems a bit weird to have some hardcoded strings and some resource-driven.   Either way is likely fine, but we may want to consider consistency.